src/test/stub/len_vec.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+extern crate libc;
+
+use std::mem;
+use std::ptr::{write, read};
+
+// Abstraction which tracks only the length and the last element of the vector.
+pub struct Vec<T> {
+    len: usize,
+    last: Option<*mut T>,
+}
+
+impl<T> Vec<T> {
+    fn drain(&mut self) {
+        self.len = 0;
+        self.last = None;
+    }
+
+    fn update_last(&mut self, elem: T) {
+        if self.len == 0 {
+            let elem_size = mem::size_of::<T>();
+            let ptr = unsafe { libc::malloc(elem_size) as *mut T };
+            self.last = Some(ptr);
+        }
+
+        unsafe {
+            write(self.last.unwrap().offset(0), elem);
+        }
+    }
+
+    pub fn new() -> Vec<T> {
+        Vec {
+            len: 0,
+            last: None
+        }
+    }
+
+    pub fn with_capacity(_: usize) -> Self {
+        Vec {
+            len: 0,
+            last: None
+        }
+    }
+
+    pub fn push(&mut self, elem: T) {
+        if self.len == 0 {
+            let elem_size = mem::size_of::<T>();
+            let ptr = unsafe { libc::malloc(elem_size) as *mut T };
```

Comment:
This doesn't ensure that the malloced thing has basic validity constraints. We need a way to talk about that. Its sound, but will almost certainly cause problems during verification.

Reply:
Proposal: have a `trait NondetItem<T>`

Reply:
Or, for now take advantage of assumptions.rs, and keep `Option<Box<T>>`, and then say `Box::new(__nondet::<T>())`